cluster-autoscaler/provisioningrequest/service/service.go

```diff
@@ -0,0 +1,72 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package provreqservice
+
+import (
+	"fmt"
+
+	"k8s.io/autoscaler/cluster-autoscaler/provisioningrequest/provreqwrapper"
+	"k8s.io/autoscaler/cluster-autoscaler/provisioningrequest/service/v1beta1client"
+	"k8s.io/client-go/rest"
+)
+
+// ProvisioningRequestService represents the service that is able to list,
+// access and delete different Provisioning Requests.
+type ProvisioningRequestService struct {
+	provReqV1Beta1Client *v1beta1client.ProvisioningRequestClient
+}
```

Comment:
Maybe it would be worth to consider testing `Service` instead of the `Client`. You could do this by defining the `provReqClient` interface here, and then having the `v1beta1client.ProvisioningRequestClient` implement it. This way you would be able to mock the client in the test and test the Service functionality well, without worrying about the client internal structure.

Reply:
I don't expect another implementation of v1beta1client.ProvisioningRequestClient (except one for test), however we might have later clients for v1 and v1beta1 that probably would require different clients (and probably different interfaces in case we want them). Even if we are going to introduce an interface, we'll need to put it in a separate package to avoid cycle dependancy.
I think in current case it's overengineering.